results-processor/main.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+import re
+import shutil
+import tempfile
+
+import flask
+from google.cloud import storage
+
+import wptreport
+
+
+APPENGINE_INTERNAL_IP = '10.0.0.1'
+
+
+app = flask.Flask(__name__)
+
+
+@app.route('/api/results/process', methods=['POST'])
+def task_handler():
+    if flask.request.remote_addr != APPENGINE_INTERNAL_IP:
+        return 'External requests not allowed', 403
+
+    params = flask.request.form
+    # Mandatory fields:
+    # uploader = params['uploader']
+    gcs_path = params['gcs']
+    result_type = params['type']
+    # TODO(Hexcles): Support multiple results.
+    assert result_type == 'single'
+    # Optional fields (to be deprecated once everyone has embedded metadata):
+    browser = params.get('browser')
+    revision = params.get('revision')
+
+    match = re.match(r'/([^/]+)/(.*)', gcs_path)
+    assert match
```

Comment:
Forgive my lack of Python, but what is the outcome of a failed assert?

Reply:
This will throw a HTTP 500, and traceback will be logged.

This assert really shouldn't fail as the GCS path is generated by the receiver. If this fails, there's a programming error, which is why this is an assert instead of some more elegant error handling.